apps/api/src/app.module.ts

```diff
@@ -2,35 +2,42 @@ import { join } from "path";
 import { ApolloDriver, ApolloDriverConfig } from "@nestjs/apollo";
 import { Module } from "@nestjs/common";
 import { ConfigModule, ConfigService } from "@nestjs/config";
-import { APP_PIPE } from "@nestjs/core";
+import { APP_GUARD } from "@nestjs/core";
 import { GraphQLModule } from "@nestjs/graphql";
 import { AppResolver } from "./app.resolver";
 import { AppService } from "./app.service";
+import { AuthenticatedGuard } from "./common/guards/auth.guard";
 import { DrizzleModule } from "./drizzle/drizzle.module";
 import { AuthModule } from "./modules/auth/auth.module";
+import { UsersModule } from "./modules/users/users.module";
 
 @Module({
     imports: [
         ConfigModule.forRoot({
             isGlobal: true,
+            envFilePath: "../../../.env.backend",
         }),
         GraphQLModule.forRoot<ApolloDriverConfig>({
             driver: ApolloDriver,
             playground: true,
             sortSchema: true,
             autoSchemaFile: join(process.cwd(), "schema.graphql"),
         }),
-        DrizzleModule.forRootAsync({
-            imports: [ConfigModule],
-            inject: [ConfigService],
-            useFactory: (configService: ConfigService) => ({
-                // TODO: add validation for env vars
-                url: configService.get<string>("DATABASE_URL") as string,
-                database: configService.get<string>("POSTGRES_DB") as string,
-            }),
+        DrizzleModule.forRoot({
+            url: String(process.env.DATABASE_URL),
+            database: String(process.env.POSTGRES_DB),
         }),
         AuthModule,
+        UsersModule,
+    ],
+    controllers: [],
+    providers: [
+        AppService,
+        AppResolver,
+        {
+            provide: APP_GUARD,
+            useClass: AuthenticatedGuard,
```

Comment:
no it's more convenient this way since not many pages of the application are public but few of them
this is why it's global

Reply:
@G0maa we have a conflict of opinions

Reply:
it's fine on my side, however, I prefer specific specifiers rather than global ones

Reply:
nop, I don't prefer it since it will be too common 
I will do the same as you did when it's specific to less than half of the endpoints but the access will be private for mostly all the endpoints here so why to annotate each class !